sql/table.go

```diff
@@ -273,12 +272,20 @@ func encodeSecondaryIndexes(tableID structured.ID, indexes []structured.IndexDes
 	var secondaryIndexEntries []indexEntry
 	for _, secondaryIndex := range indexes {
 		secondaryIndexKeyPrefix := encodeIndexKeyPrefix(tableID, secondaryIndex.ID)
-		secondaryIndexKey, err := encodeIndexKey(secondaryIndex, colMap, values, secondaryIndexKeyPrefix)
+		secondaryIndexKey, err := encodeIndexKey(secondaryIndex.ColumnIDs, colMap, values, secondaryIndexKeyPrefix)
 		if err != nil {
 			return nil, err
 		}
 
 		if secondaryIndex.Unique {
+			// Append the primaryIndexKeySuffix if any column is NULL.
+			for _, id := range secondaryIndex.ColumnIDs {
+				if _, ok := colMap[id]; !ok {
```

Comment:
Yeah, the check is wrong. I've fixed it; I think I prefer this to having `encodeIndexyKey` return a boolean since it makes the signature quite wide. It would be slightly more performant though, so I'm open to changing it if you feel strongly.